mypy/build.py

```diff
@@ -959,16 +959,29 @@ def find_modules_recursive(module: str, lib_path: List[str]) -> List[BuildSource
         return []
     result = [BuildSource(module_path, module, None)]
     if module_path.endswith(('__init__.py', '__init__.pyi')):
-        for item in os.listdir(os.path.dirname(module_path)):
+        # Subtle: this code prefers the .pyi over the .py if both
+        # exists, and also prefers packages over modules if both x/
+        # and x.py* exist.  How?  We sort the directory items, so x
+        # comes before x.py and x.pyi.  But the preference for .pyi
+        # over .py is encoded in find_module(); even though we see
+        # x.py before x.pyi, find_module() will find x.pyi first.  We
+        # use hits to avoid adding it a second time when we see x.py.
```

Comment:
Note that the final word here should be 'x.pyi' (since that's the one we see second). I've committed this fix without code review.